ducktape/tests/runner.py

```diff
@@ -73,10 +73,23 @@ def run_all_tests(self):
 
             # Create single testable unit and corresponding test result object
             self.current_test_context = test_context
+            result = TestResult(self.current_test_context)
+
+            if self.current_test_context.ignore:
+                # Skip running this test, but keep track of the fact that we ignored it
```

Comment:
Yeah, the reason I mentioned this is because we actually just ran into a case sort of like this today. The test is failing and we have a proposed fix which we're pretty confident of, but it wouldn't hurt to ignore the results for a few days to make sure it really is fixed.

In this case, the bug was tracked down without too much trouble, but for a transient issue that only occurs occasionally, it's useful to be able to add debugging code or settings specifically in the test (or, e.g., more verbose logging) and still run the tests.

I think they both potentially make sense. With unit tests, you usually end up with fewer of these transient errors (at least in most projects...), so that mode isn't as important. With system tests, you often need to collect enough evidence to figure out what's going wrong. We can leave this as is and come up with another decorator or a flag for this one if we want to support both approaches.